### PR TITLE
Explicitly permit starting over

### PR DESCRIPTION
**Pull request by martinthomson**
Generating a new cryptographic handshake message is a fine response to
Retry or Version Negotiation.  It's sometimes necessary, but even when
it is not, it avoids having to worry about 0-RTT packet numbers.

Followup for #1498.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -806,7 +806,8 @@ A client MUST NOT reset the packet number it uses for 0-RTT packets.  The keys
 used to protect 0-RTT packets will not change as a result of responding to a
 Retry or Version Negotiation packet unless the client also regenerates the
 cryptographic handshake message.  Sending packets with the same packet number in
-that case would compromise the packet protection for all 0-RTT packets.
+that case could compromise the packet protection for all 0-RTT packets because
```
Comment (ekr) [inline draft-ietf-quic-transport.md]: Why the change from "would" to "could". Could seems a lot more likely.

Comment (martinthomson) [reply draft-ietf-quic-transport.md]: It's not guaranteed - if the exact same data is sent again, then there is no compromise.  And, from what I know of some implementations, that might even work.  But it's a long shot.  I'll tweak it to change emphasis.
